### PR TITLE
[wip] Tweak cgu partitioning

### PR DESCRIPTION
**Pull request by wesleywiser**
r? @ghost

### CODE REVIEW

**Comment by andjo403** [comment]
as the server that is running perf do not have that many threads I think that the biggest gains is from getting functions calling the same inline marked functions in the same CGU as described in https://github.com/rust-lang/rust/pull/65281#issuecomment-541684092.
I still think that this is a good thing to do as it gives better estimates of the functions size. but to be sure to get better perf there needs to be some better algorithm then select the two smallest CGUs and merge them as there sometimes is better to merge CGUs that have many functions in common.
as this can also go the other way and give longer compile time if you are unlucky with the merging

**Comment by andjo403** [comment]
also this can be part of the solution for https://github.com/rust-lang/rust/issues/69382

**Comment by andjo403** [comment]
Remember that there is also a terminator per basic block that is not part of the statements.

**Comment by bjorn3** [comment]
There are absolutely no changes in performance.

File: src/librustc_ty/ty.rs [hunk 1]
```diff
@@ -299,7 +300,23 @@ fn instance_def_size_estimate<'tcx>(
...
+                        .filter(|stmt| match stmt.kind {
+                            StatementKind::StorageLive(_)
+                            | StatementKind::StorageDead(_)
+                            | StatementKind::AscribeUserType(..)
+                            | StatementKind::FakeRead(..)
+                            | StatementKind::Retag(..)
+                            | StatementKind::Nop => false,
+                            _ => true,
+                        })
+                        .count()
```
Comment (andjo403) [inline src/librustc_ty/ty.rs:317]: shall there maybe be something like https://github.com/rust-lang/rust/blob/963bf528292d8f97104515e32908e30c2467b6a8/src/librustc_mir/transform/inline.rs#L319-L373
or at least a +1 so the terminator is counted in some way

Comment (andjo403) [reply src/librustc_ty/ty.rs:317]: can maybe be simplified to only add extra for landingpads(10 llvm ir instructions) and resume (9 llvm ir instructions)

**Pull request closed without merging**